shared/teams/invite-by-contact/container.tsx

```diff
@@ -0,0 +1,264 @@
+import * as Constants from '../../constants/teams'
+import * as Contacts from 'expo-contacts'
+import * as Container from '../../util/container'
+import * as React from 'react'
+import * as SettingsConstants from '../../constants/settings'
+import * as SettingsGen from '../../actions/settings-gen'
+import * as TeamsGen from '../../actions/teams-gen'
+import {e164ToDisplay} from '../../util/phone-numbers'
+import {NativeModules} from 'react-native'
+import {TeamRoleType} from '../../constants/types/teams'
+import logger from '../../logger'
+
+import {ContactProps, ContactRowProps, InviteByContact} from '.'
+
+type OwnProps = Container.RouteProps<{teamname: string}>
+
+// for sorting
+const strcmp = (a, b) => (a === b ? 0 : a > b ? 1 : -1)
+const compareContacts = (a: ContactProps, b: ContactProps): number => {
+  if (a.name === b.name) {
+    return strcmp(a.value, b.value)
+  }
+  return strcmp(a.name, b.name)
+}
+
+const fetchContacts = async (): Promise<[Array<ContactProps>, string]> => {
+  const contacts = await Contacts.getContactsAsync({
+    fields: [
+      Contacts.Fields.Name,
+      Contacts.Fields.PhoneNumbers,
+      Contacts.Fields.Emails,
+      Contacts.Fields.ImageAvailable,
+      Contacts.Fields.Image,
+    ],
+  })
+
+  let defaultCountryCode: string = ''
+  try {
+    defaultCountryCode = await NativeModules.Utils.getDefaultCountryCode()
```

Comment:
We should dispatch a `loadedUserCountryCode` and get this from `state.settings.contacts.userCountryCode`.

Reply:
but we also want this to work when someone hasn't enabled contact syncing. Should it try to do both? First pull from state, and if it fails, use NativeModules? This complicates the importing in this component

Reply:
Yes, I think it should try to do both. 
```
let defaultCountryCode = Container.useSelector(s => s.settings.contacts.userCountryCode)
React.useEffect(() => {
  if (!defaultCountryCode) {
    ... fetch ... createLoadedUserCountryCode
  }
})
```